pandas/core/apply.py

```diff
@@ -1330,7 +1330,7 @@ def relabel_result(
     >>> funcs = {"A": ["max"], "C": ["max"], "B": ["mean", "min"]}
     >>> columns = ("foo", "aab", "bar", "dat")
     >>> order = [0, 1, 2, 3]
-    >>> _relabel_result(result, func, columns, order)  # doctest: +SKIP
+    >>> relabel_result(result, funcs, columns, order)
```

Comment:
this one is throwing an error in CI:
```python
=================================== FAILURES ===================================
__________________ [doctest] pandas.core.apply.relabel_result __________________
1324     order: New order for relabelling
1325 
1326     Examples:
1327     ---------
1328     >>> result = DataFrame({"A": [np.nan, 2, np.nan],
1329     ...       "C": [6, np.nan, np.nan], "B": [np.nan, 4, 2.5]})  # doctest: +SKIP
1330     >>> funcs = {"A": ["max"], "C": ["max"], "B": ["mean", "min"]}
1331     >>> columns = ("foo", "aab", "bar", "dat")
1332     >>> order = [0, 1, 2, 3]
1333     >>> relabel_result(result, funcs, columns, order)
UNEXPECTED EXCEPTION: NameError("name 'result' is not defined")
Traceback (most recent call last):
  File "/home/runner/micromamba/envs/test/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest pandas.core.apply.relabel_result[4]>", line 1, in <module>
NameError: name 'result' is not defined
```
I think you need to remove `# doctest: +SKIP` from a few lines above

Reply:
Fixed the error!

Reply:
Thanks @kkangs0226! ... Is there any chance that you could run this command? : 

./scripts/validate_docstrings.py pandas.core.apply.relabel_result

Reply:
Here is the output, it still flags as failed example but I think @MarcoGorelli has mentioned that this error is unrelated. It seems to be one of F821 undefined errors. 
![image](https://user-images.githubusercontent.com/66008784/221407182-b6f44b1d-6b6a-4194-95bc-4d241fefd269.png)

Reply:
sorry I meant that the `Ubuntu / Numpy Dev (pull_request)` CI job's failure is unrelated

this one looks related, you may need to put something like `from pandas.core.apply import relabel_result` at the top of the doctest

Reply:
After running the example, I get a ```KeyError: "None of [Index([-1], dtype='int64')] are in the [index]"``` error. I have looked into it but am not sure how to fix the example, please give me some guidance on how I can fix the example!

Reply:
thanks for trying - sure, I'll take a look tomorrow, hopefully there's a simple fix

Reply:
The docstring of `relabel_result` has many errors. I see 16: 
The summary should be in one line, the parameters and even the subtitles are incorrect. (They shouldn't have ":", it should be `Examples`, not `Examples:`) etc. 

But at the beginning of the function it's written "Internal function...", and this is not on the documentation website.
Should we test internal functions docstrings?! @MarcoGorelli 

Thank you @kkangs0226... We'll get there. :-)

Reply:
I don't think we should fix the whole docstring here. Let's merge this when all the problems to the original PR scope are fixed, and open follow up PRs for anything else we want to fix.

@kkangs0226 can you provide the whole traceback for the error please? With only the error and the line you shared I don't know what the problem can be, and more context would be helpful.

Reply:
```Line 21, in pandas.core.apply.relabel_result
Failed example:
    relabel_result(result, funcs, columns, order)
Exception raised:
    Traceback (most recent call last):
      File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest pandas.core.apply.relabel_result[6]>", line 1, in <module>
        relabel_result(result, funcs, columns, order)
      File "/Users/kingbob/Desktop/NUS/3281:3282/pandas/pandas/core/apply.py", line 1377, in relabel_result
        s = s[col_idx_order]
      File "/Users/kingbob/Desktop/NUS/3281:3282/pandas/pandas/core/series.py", line 968, in __getitem__
        return self._get_with(key)
      File "/Users/kingbob/Desktop/NUS/3281:3282/pandas/pandas/core/series.py", line 1003, in _get_with
        return self.loc[key]
      File "/Users/kingbob/Desktop/NUS/3281:3282/pandas/pandas/core/indexing.py", line 1100, in __getitem__
        return self._getitem_axis(maybe_callable, axis=axis)
      File "/Users/kingbob/Desktop/NUS/3281:3282/pandas/pandas/core/indexing.py", line 1329, in _getitem_axis
        return self._getitem_iterable(key, axis=axis)
      File "/Users/kingbob/Desktop/NUS/3281:3282/pandas/pandas/core/indexing.py", line 1269, in _getitem_iterable
        keyarr, indexer = self._get_listlike_indexer(key, axis)
      File "/Users/kingbob/Desktop/NUS/3281:3282/pandas/pandas/core/indexing.py", line 1459, in _get_listlike_indexer
        keyarr, indexer = ax._get_indexer_strict(key, axis_name)
      File "/Users/kingbob/Desktop/NUS/3281:3282/pandas/pandas/core/indexes/base.py", line 5851, in _get_indexer_strict
        self._raise_if_missing(keyarr, indexer, axis_name)
      File "/Users/kingbob/Desktop/NUS/3281:3282/pandas/pandas/core/indexes/base.py", line 5910, in _raise_if_missing
        raise KeyError(f"None of [{key}] are in the [{axis_name}]")
    KeyError: "None of [Index([-1], dtype='int64')] are in the [index]"```
Here is the entire traceback, hope it is able to provide more context.

Reply:
ok, got it - I'll post a new comment showing how I'd go about fixing this